README.md

```diff
@@ -1 +1,235 @@
+    GA: 'UA-126263119-1'
+}
+</script>
+```
+
+#### `index.html` must never be cached
+
+To allow for web application environments to be changed instantly, `index.html` must [never be cached by the browser or a public cache that cannot be purged on-demand:](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#Preventing_caching)
+
+`cache-control: no-store`
+
+##### _Example_
+
+The `index.html` of most single-page web applications is typically a small document. By including versioned references to the web application static assets and setting the environment variables it is effectively a deployment manifest:
+
+```html
+<!doctype html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <title>My App</title>
+    <base href="/">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="icon" type="image/x-icon" href="favicon.ico">
+</head>
+    <body>
+
+        <!-- environment variables -->
+        <script>
+        env = {
+            API: 'https://api.myapp.com'
+        }
+        </script>
+
+        <!-- application binding -->
+        <app-root></app-root>
+
+        <!-- fully-qualified static assets -->
+        <script src="https://assets.myapp.com/apps/bae1407/main.js" type="text/javascript"></script>
+
+    </body>
+</html>
+```
+
+## Workflow
+
+The _Immutable Web App_ separate release tasks from build tasks into two distinct workflows.
+
+### Building
+
+The codebase of an _Immutable Web App_ has the responsibility of building static assets and publishing them to a static web server. Each state of the codebase can be represented by a set of static assets at a unique location. Not every state of the codebase needs to be published, but no single state of the codebase should ever need to be published more than once.
+
+Generally the codebase is a source controlled code repository that is capable of building, versioning, and publishing static assets to a static web server.
+
+_An example of this might be:_
+
+An __Angular__ project hosted in a __GitHub__ repo. The repo is integrated with __TravisCI__ to build and version assets when commits are pushed to the master branch. Versioned assets are published to a unique location in an __AWS S3 bucket__.
+
+### Releasing
+
+The `index.html` files are managed independently of the codebase and they serve as a manifest for each environment. They should be considered configuration files and managed accordingly. Additionally, There needs to be a mechanism for modifying or replacing the `index.html` in each web application environment. The act of changing an `index.html` is effectively a deployment.
+
+_An example of this might be:_
+
+A set of `index.html` files, one for each environment, hosted in a __Github__ repo. The repo is integrated with __TravisCI__ to publish an `index.html` file when it is modified to an __AWS S3 bucket__. There is an `index.html` and S3 bucket dedicated to each web application environment.
```

Comment:
This reads as though there needs to be a separate S3 bucket for every environment.  That isn't necessarily required, correct?  I understand you are just trying to lay out an example though, just wondering if you need to be that prescriptive with the example.